.github/workflows/ci.yml

```diff
@@ -0,0 +1,108 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: XNIO CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  build-all:
+    name: Compile (no tests) with JDK 8
+    runs-on: ubuntu-latest
+    steps:
+    - uses: n1hility/cancel-previous-runs@v2
+      with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+    - uses: actions/cache@v1
+      with:
+        path: ~/.m2/repository
+        key: m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+            m2-
+    - uses: actions/checkout@v2
+    - name: Host information
+      run: netstat -anp
+      shell: bash
+    - name: Set up JDK 8
+      uses: joschi/setup-jdk@v2
+      with:
+        java-version: 8
+    - name: Build
+      run: mvn -U -B -fae -DskipTests -Dfindbugs clean install
+    - name: Tar Maven Repo
+      shell: bash
+      run: tar -czf maven-repo.tgz -C ~ .m2/repository
+    - name: Persist Maven Repo
+      uses: actions/upload-artifact@v1
+      with:
+        name: maven-repo
+        path: maven-repo.tgz
+    - uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: surefire-reports-build
+        path: '**/surefire-reports/*.txt'
+  test-matrix:
+    name: ${{ matrix.module }}-${{ matrix.openjdk_impl}}-${{ matrix.jdk }}-${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    needs: build-all
+    strategy:
+      fail-fast: false
+      matrix:
+        module: [api, nio-impl]
+        os: [ubuntu-20.04, windows-latest]
+        jdk: [8, 11]
+        openjdk_impl: [openjdk, openj9]
+        exclude:
+          - os: ubuntu-20.04
+            jdk: 11
+          - os: windows-latest
+            jdk: 11
```

Comment:
This is probably the most aggressive test-matrix I've ever seen. :slightly_smiling_face:

We'd also want `macos` on there as well though in this case.  Also it seems we're excluding JDK 11 for both of the possible OSes... is there a consistent test failure?

Reply:
Yeah, I can add macos, no problem, though be aware I think it might bill your quota at 10x the rate of linux, and I haven't tried using it yet :)

The JDK 11 issue is that apparently we didn't have jdk 11 jobs in TC for this (I added two yesterday to compare) and they fail:

https://ci.wildfly.org/buildConfiguration/Xnio_3x_WindowsJdk11testJob/214809
https://ci.wildfly.org/buildConfiguration/Xnio_3x_Linux_2/214808

So these can be enabled in here at any point along with 14 etc, but they'll fail, so I thought easier to exclude now and then re-enable in a PR when someone has more time to look.

Reply:
OK.  The XNIO test suite as a whole is a bit of a train wreck so don't feel too bad if everything isn't green from the start.  It's something that has been on my TODO for quite a long time...